### PR TITLE
Automatically generate the audit events table

### LINKED ISSUE #10350: [v.8.0] /docs/pages/setup/reference/audit.mdx
## Details

The list of "all possible audit events" is out of date and incomplete.

Either decide that we should provide a complete list of all events (which is what I would prefer), or remove the list entirely.

### Category

- Improve Existing

### PR DESCRIPTION
**Pull request by ptgott**
Automatically generate the audit events table

Ensure that we have a complete table of audit events in the docs by
generating the table automatically from the source.

This change assumes that all `AuditEvent` implementations include a
`Metadata` field with the value type `Metadata` from the
`api/types/events` package. It parses all Go files in the
`gravitational/teleport` source, then traverse the resulting ASTs twice
to:

- Build a map of `Metadata` types based on all assignments of an
  `events.Metadata` to a `Metadata` field.
- Find all declarations with names that match a key in the map in order
  to extract their comments.

This also adds a Make target for auto-generating docs partials. We can
use this for any additional docs generation.

This change also ensures that the audit event table is sorted
alphabetically, and bolds the first column of the table.

Note that this change means that we lose some of the manually added
information from the original table, including some JSON fields and
whether a given event applies only to Enhanced Session Recording. We
will need to find a way to gather this information programmatically.

The original table only provided this information for a small subset of
the audit events, though, and it's arguably more valuable to include
names and descriptions of all audit events than more complete
information for only some of them.

Finally, this change refines the comments above audit event declarations
to improve the language of descriptions in the table.

### CODE REVIEW

**Comment by ptgott** [comment]
I still need to:

- [ ] See if there's a more reliable way to extract audit events from the source. The `events.AuditEvent` and `events.Emitter` interfaces don't require a `Metadata` field, so it's possible that the current implementation ignores current/future audit events in the source.
- [ ] Refine/QA the results. For example, there are extraneous line breaks within some table rows. It would also be nice to sort the rows alphabetically by the event type.
- [ ] Add unit tests

File: assets/docs-gen/go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module docs-gen
+
+go 1.17
```
Comment (reedloden) [inline assets/docs-gen/go.mod:3]:
```suggestion
go 1.18
```

**Comment by ptgott** [comment]
@klizhentas Regarding your recommendation to add a linter to ensure the source is structured the way this auto-generation code expects, should I add that within this PR, or create a separate PR later? Thanks!

**Comment by ptgott** [comment]
I am going to close this PR while I develop a linter and open another one when I have something ready to share

**Pull request closed without merging**